src/arvcamera.c

```diff
@@ -1225,43 +1225,53 @@ arv_camera_set_trigger (ArvCamera *camera, const char *source, GError **error)
 	gboolean has_frame_start = FALSE;
         gboolean has_frame_burst_start = FALSE; /* Hikrobot, Basler devices */
 	gboolean has_acquisition_start = FALSE; /* Smartek devices */
-        const char **triggers = NULL;
-        guint n_triggers = 0;
-        unsigned int i;
+        gboolean has_trigger_selector = FALSE;
 
 	g_return_if_fail (ARV_IS_CAMERA (camera));
 	g_return_if_fail (source != NULL);
 
 	if (arv_camera_is_feature_available (camera, "AcquisitionFrameRateEnable", NULL))
                 arv_camera_set_boolean (camera, "AcquisitionFrameRateEnable", FALSE, &local_error);
 
-	triggers = arv_camera_dup_available_enumerations_as_strings (camera, "TriggerSelector", &n_triggers,
-								     &local_error);
-
-        for (i = 0; i < n_triggers && local_error == NULL; i++) {
-                arv_camera_set_string (camera, "TriggerSelector", triggers[i], &local_error);
-                if (local_error == NULL) {
-                        if (g_strcmp0 (triggers[i], "FrameStart") == 0)
-                                has_frame_start = TRUE;
-                        else if (g_strcmp0 (triggers[i], "FrameBurstStart") == 0)
-                                has_frame_burst_start = TRUE;
-                        else if (g_strcmp0 (triggers[i], "AcquisitionStart") == 0)
-                                has_acquisition_start = TRUE;
-                        arv_camera_set_string (camera, "TriggerMode", "Off", &local_error);
+        has_trigger_selector = arv_camera_is_feature_available(camera, "TriggerSelector", &local_error);
+
+        if (has_trigger_selector == TRUE) {
+                const char **triggers = NULL;
+                guint n_triggers = 0;
+                unsigned int i;
+
+                triggers = arv_camera_dup_available_enumerations_as_strings (camera, "TriggerSelector", &n_triggers,
```

Comment:
I did not change that part (only indentation) but this could be replaced by `arv_camera_dup_available_triggers`